server/game/costs/RevealCost.js

```diff
@@ -1,14 +1,16 @@
+const GameActions = require('../GameActions');
+
 class RevealCost {
     constructor() {
         this.name = 'reveal';
     }
 
     isEligible(card) {
-        return card.location === 'hand';
+        return ['hand', 'plot deck', 'shadows'].includes(card.location);
```

Comment:
For consistency, this would probably be better expressed using the `RevealCards.allow` method itself, so that we're not repeating implementations.

Reply:
This is slightly different to `RevealCards.allow`, for revealing as a cost will require a hidden location which you have access to, but your opponents do not (your hand, plot deck or shadows). For example, you cannot reveal cards in your deck as a cost without them already being revealed (thus a cost being impossible).

Reply:
On second thought, it's probably better to keep this as flexible as possible. Eg. there _could_ be an ability which says to "Search your deck for a card. Reveal that card to [do something unusual]", so I'll implement your thoughts now.